plugins/modules/cloud/misc/proxmox_interfaces.py

```diff
@@ -0,0 +1,414 @@
+        type: bridge
+        address: 10.189.5.37
+        netmask: 255.255.255.0
+        mtu: 5555
+      - name: vmrb56
+        type: bridge
+        address: 192.168.5.4
+        netmask: 255.255.255.0
+        comments: 'This is a new VM bridge with the host inside'
+    state: present
+
+- name: Delete a list of interfaces from node01 and try to revert on error
+  community.general.proxmox_interfaces:
+    api_user: root@pam
+    api_password: secret
+    api_host: proxmoxhost
+    node: node01
+    config:
+      - name: vmbr34
+        state: absent
+      - name: vmbr48
+        state: absent
+    apply: true
+    revert_on_error: true
+'''
+
+RETURN = '''
+before:
+  description: The configuration as structured data before module execution
+  returned: always
+  type: list
+  sample:
+    - '{some example JSON here}'
+after:
+  description: The configuration as structured data after module completion
+  returned: when changed
+  type: list
+  sample:
+    - '{some different JSON data here}'
+upid:
+  description: UPID of the svreload:networking task
+  returned: when changed and reload_interfaces is enabled
+  type: str
+  sample: 'UPID:node01:00003098:16036ACE:6216F174:srvreload:networking:root@pam:'
+errors:
+  description: Errors that occurred and where ignored
+  returned: when failed
+  type: list
+  sample:
+    - list of errors
+'''
+
+import traceback
+
+PROXMOXER_IMP_ERR = None
+try:
+    from proxmoxer import ProxmoxAPI
+    HAS_PROXMOXER = True
+except ImportError:
+    HAS_PROXMOXER = False
+    PROXMOXER_IMP_ERR = traceback.format_exc()
+
+from ansible_collections.community.general.plugins.module_utils.proxmox import (
+    ProxmoxAnsible, proxmox_auth_argument_spec)
+from ansible_collections.community.general.plugins.module_utils.proxmox_interfaces import (
+    get_nics, delete_nic, create_nic, reload_interfaces, rollback_interfaces,
+    update_nic, proxmox_map_interface_args, proxmox_interface_argument_spec, check_doublicates)
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    nic_args = proxmox_interface_argument_spec()
+    module_args = proxmox_auth_argument_spec()
+    network_args = dict(
+        config=dict(type='list', elements='dict',
+                    required=True, options=nic_args),
+        node=dict(type='str', required=True),
+        ignore_errors=dict(type='bool', default=False),
+        apply=dict(type='bool', default=True),
+        revert_on_error=dict(type='bool', default=False),
+        state=dict(type='str', choices=[
+                   'present', 'reloaded'], default='present')
+    )
+    module_args.update(network_args)
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        required_together=[('api_token_id', 'api_token_secret'),
+                           ('api_user', 'api_password')],
+        required_one_of=[('api_password', 'api_token_id')],
+        required_if=[('state', 'present', ('config',))],
+        supports_check_mode=False,
```

Comment:
I'd suggest to implement check mode, in the sense that the module informs whether it would make changes or not. From the code below, it looks rather simple to implement.